src/Bicep.Core/LanguageConstants.cs

```diff
@@ -69,7 +69,7 @@ public static TypeSymbol CreateParameterModifierType(TypeSymbol primitiveType, T
 
         private static IEnumerable<TypeProperty> CreateParameterModifierProperties(TypeSymbol primitiveType, TypeSymbol allowedValuesType)
         {
-            if (ReferenceEquals(primitiveType, String) || ReferenceEquals(primitiveType, Object))
+            if (ReferenceEquals(primitiveType, String) || ReferenceEquals(primitiveType, Object) || ReferenceEquals(primitiveType, Any))
```

Comment:
Are the checks for `any` just to handle the scenario where we haven't been able to parse the type properly? If so, is there a different type we could use to make this clearer (I'm thinking ErrorTypeSymbol potentially)? Loosening the strictness of this method feels like it could make it easier for future bugs to slip in.

Reply:
(if not, mind adding some comments to this file just to explain the check for any?)

Reply:
Will do...

Reply:
Added in next update.

Reply:
Btw, I also looked into using a different type than `any` for this and other similar places. If we use an error type, it short circuit s the type checks inside the object. I would rather we produce the errors from inside the modifier than make them suddenly appear when the user fixes the parse error on the type. Alternatively, we could prevent the short circuiting somehow for errorless error types (don't want a type error because there's already a parse error for the same thing), but I don't think it's worth it for an edge case like this.